File: crates/nil-payload/src/infrastructure/prefecture.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::continent::Coord;
use nil_core::infrastructure::building::prefecture::PrefectureBuildOrderRequest;
use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AddPrefectureBuildOrderRequest {
  pub world: WorldId,
  pub request: PrefectureBuildOrderRequest,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CancelPrefectureBuildOrderRequest {
  pub world: WorldId,
  pub coord: Coord,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPrefectureBuildCatalogRequest {
  pub world: WorldId,
  pub coord: Coord,
}

File: app/src/types/request/infrastructure/prefecture.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { WorldId } from '@/types/core/world';
import type { Coord } from '@/types/core/continent';
import type { PrefectureBuildOrderRequest } from '@/types/core/infrastructure/prefecture';

export interface AddPrefectureBuildOrderRequest {
  readonly world: WorldId;
  readonly request: PrefectureBuildOrderRequest;
}

export interface CancelPrefectureBuildOrderRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}

export interface GetPrefectureBuildCatalogRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}
